chain/client/src/stateless_validation/chunk_validator.rs

```diff
@@ -341,6 +352,10 @@ fn validate_source_receipt_proofs(
     // Iterate over blocks between last_chunk_block (inclusive) and last_last_chunk_block (exclusive),
     // from the newest blocks to the oldest.
     for block in receipt_source_blocks {
+        if block.header().is_genesis() {
+            assert_eq!(receipt_source_blocks.len(), 1);
+            break;
+        }
```

Comment:
nit: Maybe it would be nicer to have an explicit condition: `if witnes for genesis { source receipts should be empty}`. I feel that it would be easier to reason about than the current solution.

Reply:
This assert is not part of validation logic, it ensures that our chain is in the valid state (`receipt_source_blocks` is not calculated using our own chain state, not state witness). I will replace it with proper error handling so you feel better about it 😉 

> Maybe it would be nicer to have an explicit condition

It is a bit more code, but indeed should be easier to understand, done